### PR TITLE
fix bugs in dual

### PR DESCRIPTION
**Pull request by bpiwowar**

### CODE REVIEW

**Review by Mosakana — APPROVED** [review]
add merge function for TextsRepresentationOutput class.

File: src/xpmir/neural/dual.py [hunk 1]
```diff
@@ -67,6 +67,13 @@ class Dense(DualVectorScorer[QueriesRep, DocsRep]):
     """A scorer based on a pair of (query, document) dense vectors"""
 
     def score_product(self, queries, documents, info: Optional[TrainerContext] = None):
+        # logger.info(f'info : {info} \n')
```
Comment (bpiwowar) [inline src/xpmir/neural/dual.py:70]: Please remove

File: src/xpmir/neural/dual.py [hunk 2]
```diff
@@ -170,6 +177,13 @@ def __call__(self, info: TrainerContext, queries, documents):
         # Assumes that all weights are positive
         assert info.metrics is not None
 
+#####################################################
```
Comment (bpiwowar) [inline src/xpmir/neural/dual.py:180]: remove the comments

File: src/xpmir/neural/__init__.py [hunk 3]
```diff
@@ -92,7 +92,39 @@ def merge(self, objects: Union[DocsRep, QueriesRep]):
...
+            mask = list(map(lambda x: x.mask, tokenized))
+            mask = None if len(mask) == 0 else merge_mask(mask)
+
+            token_type_ids = list(filter(lambda x: x is not None, map(lambda x: x.token_type_ids, tokenized)))
+            token_type_ids = None if len(token_type_ids) == 0 else torch.cat(token_type_ids)
+
+            return TextsRepresentationOutput(torch.cat(list(map(lambda x: x.value, objects))), TokenizedTexts(tokens, ids, lens, mask, token_type_ids))
+
+            
+################################################################################
```
Comment (bpiwowar) [inline src/xpmir/neural/__init__.py:127]: remove this line

File: src/xpmir/neural/__init__.py [hunk 4]
```diff
@@ -92,7 +92,39 @@ def merge(self, objects: Union[DocsRep, QueriesRep]):
 
         if isinstance(objects[0], List):
             return list(itertools.chain(objects))
+        
+################################################################################
```
Comment (bpiwowar) [inline src/xpmir/neural/__init__.py:96]: remove this line

**Review by bpiwowar — COMMENTED** [review]
And don't forget to run `pre-commit` before commit

**Pull request merged**